.github/workflows/perf.yml

```diff
@@ -63,19 +63,19 @@ jobs:
           ref: v4.9.5
           path: ${{ env.TEST_DIR }}
       - name: Profile memory (base)
-        run: valgrind --tool=massif --massif-out-file=${{ env.BASE_MASSIF_OUT }} ${{ env.BASE_DIR }}/target/release/${{ env.BIN }} analyze --max-file-time=30 -- ${{ env.TEST_DIR }}
+        run: valgrind --tool=massif --massif-out-file=${{ env.BASE_MASSIF_OUT }} ${{ env.BASE_DIR }}/target/release/${{ env.BIN }} analyze --max-file-time=30 --hide-error-details -- ${{ env.TEST_DIR }}/src/compiler
       - name: Profile memory (head)
-        run: valgrind --tool=massif --massif-out-file=${{ env.HEAD_MASSIF_OUT }} ${{ env.HEAD_DIR }}/target/release/${{ env.BIN }} analyze --max-file-time=30 -- ${{ env.TEST_DIR }}
+        run: valgrind --tool=massif --massif-out-file=${{ env.HEAD_MASSIF_OUT }} ${{ env.HEAD_DIR }}/target/release/${{ env.BIN }} analyze --max-file-time=30 --hide-error-details -- ${{ env.TEST_DIR }}/src/compiler
```

Comment:
I'm not sure if this is a concern at this time, but something that stood out to me is that the profile steps run sequentially. If the profile steps were instead defined as separate jobs, then they will run in parallel (and appears safe to do so here). If the profile jobs run in parallel, then the `Generate report` step would also need to be defined as a job that [`needs`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds) the two profile jobs before it can run.

Reply:
That is a very nice suggestion. Profiling is a bit slow, so if they can run in parallel, it woul dbe very useful.